libtransmission/rpc-server.cc

```diff
@@ -643,6 +651,206 @@ void rpc_server_start_retry_cancel(tr_rpc_server* server)
     server->start_retry_counter = 0;
 }
 
+#ifdef WITH_OPENSSL
+struct bufferevent* bevcb(struct event_base* base, void* arg)
+{
+    struct bufferevent* r = nullptr;
+    auto ctx = reinterpret_cast<SSL_CTX*>(arg);
+    r = bufferevent_openssl_socket_new(base, -1, SSL_new(ctx), BUFFEREVENT_SSL_ACCEPTING, BEV_OPT_CLOSE_ON_FREE);
+    return r;
+}
+
+int tr_SSL_CTX_use_certificate_chain_file(SSL_CTX* ctx, char const* file)
+{
+#if OPENSSL_VERSION_NUMBER < 0x30000000
+    return SSL_CTX_use_certificate_chain_file(ctx, file);
+#else
+    /*Encountered some problem with BIO_s_file() in windows so read certs into mem first*/
```

Comment:
Is this your comment or not? If something's not working on Windows or with OpenSSL 3.0, I'd rather find a [much] shorter solution since OpenSSL 3.0+ is the new default (1.1 was EOLed last year) and I don't want to support this logic going forward.

Reply:
I tried again and there is still some problem with `BIO_s_file()` on windows, so original `SSL_CTX_use_certificate_chain_file` cannot work well. Working to find out why currently...

Reply:
similar problem described [here](https://stackoverflow.com/questions/9796402/load-certificate-chain-from-pem-file-into-stack-ofx509-using-openssl)

Reply:
the problem is that openssl is built with 'no-stdio', so it cannot open file, now it is restored and this should be simply `SSL_CTX_use_certificate_chain_file`. (It takes sooooo long time to find out this)